File: Module/PlayWays/src/main/java-room-rank/com/module/playways/room/room/comment/holder/CommentDynamicHolder.java
package com.module.playways.room.room.comment.holder;

import android.support.v7.widget.RecyclerView;
import android.view.View;

import com.common.core.avatar.AvatarUtils;
import com.common.image.fresco.BaseImageView;
import com.common.image.fresco.FrescoWorker;
import com.common.image.model.ImageFactory;
import com.common.utils.U;
import com.common.view.DebounceViewClickListener;
import com.component.busilib.view.AvatarView;
import com.facebook.drawee.drawable.ScalingUtils;
import com.facebook.drawee.view.SimpleDraweeView;
import com.module.playways.room.room.comment.adapter.CommentAdapter;
import com.module.playways.room.room.comment.model.CommentDynamicModel;
import com.module.playways.R;

public class CommentDynamicHolder extends RecyclerView.ViewHolder {
    AvatarView mAvatarIv;
    SimpleDraweeView mCommentSdv;

    int position;
    CommentDynamicModel commentModel;

    public CommentDynamicHolder(View itemView, CommentAdapter.CommentAdapterListener mCommentItemListener) {
        super(itemView);

        mAvatarIv = itemView.findViewById(R.id.avatar_iv);
        mCommentSdv = itemView.findViewById(R.id.comment_sdv);

        mAvatarIv.setOnClickListener(new DebounceViewClickListener() {
            @Override
            public void clickValid(View v) {
                if (mCommentItemListener != null && !commentModel.isFake()) {
                    mCommentItemListener.clickAvatar(commentModel.getUserInfo().getUserId());
                }
            }
        });
    }


    public void bind(int position, CommentDynamicModel commentModel) {
        this.position = position;
        this.commentModel = commentModel;

        mAvatarIv.bindData(commentModel.getUserInfo());
        FrescoWorker.loadImage(mCommentSdv, ImageFactory.newPathImage(commentModel.getDynamicModel().getBigEmojiURL())
                .setScaleType(ScalingUtils.ScaleType.CENTER_INSIDE)
                .setFitDrawable(true)
                .build());
    }
}

File: Module/PlayWays/src/main/java-room-rank/com/module/playways/room/room/comment/holder/CommentDynamicHolder.kt
package com.module.playways.room.room.comment.holder

import android.support.v7.widget.RecyclerView
import android.view.View
import com.common.core.myinfo.MyUserInfoManager

import com.common.image.fresco.FrescoWorker
import com.common.image.model.ImageFactory
import com.common.view.DebounceViewClickListener
import com.component.busilib.view.AvatarView
import com.facebook.drawee.drawable.ScalingUtils
import com.facebook.drawee.view.SimpleDraweeView
import com.module.playways.room.room.comment.adapter.CommentAdapter
import com.module.playways.room.room.comment.model.CommentDynamicModel
import com.module.playways.R

class CommentDynamicHolder(itemView: View, mCommentItemListener: CommentAdapter.CommentAdapterListener?) : RecyclerView.ViewHolder(itemView) {

    private val mAvatarIv: AvatarView = itemView.findViewById(R.id.avatar_iv)
    private val mCommentSdv: SimpleDraweeView = itemView.findViewById(R.id.comment_sdv)

    internal var position: Int = 0
    internal var commentModel: CommentDynamicModel? = null

    init {
        mAvatarIv.setOnClickListener(object : DebounceViewClickListener() {
            override fun clickValid(v: View) {
                if (commentModel?.isFake == false) {
                    mCommentItemListener?.clickAvatar(commentModel?.userInfo?.userId ?: 0)
                }
            }
        })
    }


    fun bind(position: Int, commentModel: CommentDynamicModel) {
        this.position = position
        this.commentModel = commentModel

        if (commentModel.isFake) {
            mAvatarIv.bindData(commentModel.userInfo, commentModel.fakeUserInfo?.nickName, commentModel.fakeUserInfo?.avatarUrl)
        } else {
            mAvatarIv.bindData(commentModel.userInfo, commentModel.fakeUserInfo?.nickName, commentModel.userInfo?.avatar)
        }
        FrescoWorker.loadImage(mCommentSdv, ImageFactory.newPathImage(commentModel.dynamicModel!!.bigEmojiURL)
                .setScaleType(ScalingUtils.ScaleType.CENTER_INSIDE)
                .setFitDrawable(true)
                .build())
    }
}
